controllers/articleControllers.js

```diff
@@ -0,0 +1,25 @@
+const { fetchArticleById, updateArticle } = require("../models/articleModels");
+
+exports.getArticleById = (req, res, next) => {
+  const id = req.params.article_id;
+  fetchArticleById(id)
+    .then((article) => {
+      res.status(200).send({ article });
+    })
+    .catch(next);
+};
+
+exports.patchArticle = (req, res, next) => {
+  const id = req.params.article_id;
+  const incObj = req.body;
+  fetchArticleById(id)
```

Comment:
This is a little overengineered, we don't actually need the current article data to do a PATCH req.

You're currently fetching an article but then not using the result of this request . You can remove this and instead just have:

`updateArticle(id, incObj).then(() => {
res.status(200).send({article})
})`

If the id is notValid etc. you can handle it with your error handlers :)